Observations: Attribute values have changed type from `java.lang.Long` to `java.lang.Integer`

For example: 
memory pool `used` and `max`
Has anything changed in the Java client recently?

I made a test that compared types on some potential suspects. Found the following. 
```
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=Metaspace,type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Young Generation,type=GarbageCollector.LastGcInfo.duration
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'profiled nmethods',type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Old Gen,type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=Metaspace,type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'non-profiled nmethods',type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Eden Space,type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Eden Space,type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'non-profiled nmethods',type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Eden Space,type=MemoryPool.CollectionUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Old Gen,type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'non-nmethods',type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=Compressed Class Space,type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=Compressed Class Space,type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Survivor Space,type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Old Gen,type=MemoryPool.CollectionUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Survivor Space,type=MemoryPool.CollectionUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=G1 Survivor Space,type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'non-nmethods',type=MemoryPool.PeakUsage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
--
java.lang.AssertionError: Mismatch in returned type for attribute path : java.lang:name=CodeHeap 'profiled nmethods',type=MemoryPool.Usage.committed
Expected :class java.lang.Long
Actual   :class java.lang.Integer
````

I will try to adjust type in the adapter and measure the effect

--- a/client/java/src/main/java/org/jolokia/client/J4pClient.java
+++ b/client/java/src/main/java/org/jolokia/client/J4pClient.java
@@ -347,7 +347,7 @@ public class J4pClient extends J4pClientBuilderFactory {
             if (pJsonResponse instanceof JSONObject) {
                 JSONObject errorObject = (JSONObject) pJsonResponse;
 
-                if (!errorObject.containsKey("status") || (Integer) errorObject.get("status") != 200) {
+                if (!errorObject.containsKey("status") || (Long) errorObject.get("status") != 200) {
                     throw new J4pRemoteException(null, errorObject);
                 }
             }

--- a/client/java/src/main/java/org/jolokia/client/exception/J4pRemoteException.java
+++ b/client/java/src/main/java/org/jolokia/client/exception/J4pRemoteException.java
@@ -65,7 +65,7 @@ public class J4pRemoteException extends J4pException {
     public J4pRemoteException(J4pRequest pJ4pRequest, JSONObject pJsonRespObject) {
         super(generateErrorMessage(pJ4pRequest, pJsonRespObject));
 	Object statusO = pJsonRespObject.get("status");
-        Integer statusL = statusO instanceof Integer ? (Integer) statusO : null;
+        Integer statusL = statusO instanceof Integer ? (Integer) statusO : (statusO instanceof Long ? ((Long) statusO).intValue() : null);
         status = statusL != null ? statusL : 500;
         request = pJ4pRequest;
 	response = pJsonRespObject;

--- a/client/java/src/main/java/org/jolokia/client/request/ValidatingResponseExtractor.java
+++ b/client/java/src/main/java/org/jolokia/client/request/ValidatingResponseExtractor.java
@@ -62,6 +62,9 @@ public class ValidatingResponseExtractor implements J4pResponseExtractor {
             if (o instanceof Integer) {
                 status = (Integer) o;
             }
+            if (o instanceof Long) {
+                status = ((Long) o).intValue();
+            }
         }
 
         if (!allowedCodes.contains(status)) {

--- a/client/java/src/test/java/org/jolokia/client/J4pClientTest.java
+++ b/client/java/src/test/java/org/jolokia/client/J4pClientTest.java
@@ -81,7 +81,7 @@ public class J4pClientTest {
 
         J4pClient j4p = new J4pClient(TEST_URL,client);
         J4pReadResponse resp = j4p.execute(TEST_REQUEST);
-        assertEquals(((Map<?, ?>) resp.getValue()).get("max"), 530186240);
+        assertEquals(((Map<?, ?>) resp.getValue()).get("max"), 530186240L);
     }
 
     @Test(expectedExceptions = J4pException.class,expectedExceptionsMessageRegExp = ".*JSONArray.*")

--- a/client/java/src/test/java/org/jolokia/client/request/J4pExecIntegrationTest.java
+++ b/client/java/src/test/java/org/jolokia/client/request/J4pExecIntegrationTest.java
@@ -46,9 +46,9 @@ public class J4pExecIntegrationTest extends AbstractJ4pIntegrationTest {
             j4pClient.execute(request);
             request = new J4pExecRequest(cfg,itSetup.getOperationMBean(),"fetchNumber","inc");
             J4pExecResponse resp = j4pClient.execute(request);
-            assertEquals(0,(int) resp.getValue());
+            assertEquals(0, (long) resp.getValue());
             resp = j4pClient.execute(request);
-            assertEquals(1,(int) resp.getValue());
+            assertEquals(1, (long) resp.getValue());
         }
     }
 
@@ -70,7 +70,7 @@ public class J4pExecIntegrationTest extends AbstractJ4pIntegrationTest {
                 j4pClient.execute(request);
                 fail();
             } catch (J4pRemoteException exp) {
-                assertEquals(400,exp.getStatus());
+                assertEquals(400L,exp.getStatus());
                 assertTrue(exp.getMessage().contains("IllegalArgumentException"));
                 assertTrue(exp.getRemoteStackTrace().contains("IllegalArgumentException"));
             }
@@ -154,7 +154,7 @@ public class J4pExecIntegrationTest extends AbstractJ4pIntegrationTest {
         Object[] args = new Object[] { 12,true,null, "Bla" };
         for (J4pExecRequest request : execRequests("objectArrayArg",new Object[] { args })) {
             J4pExecResponse resp = j4pClient.execute(request,"POST");
-            assertEquals(12,(int) resp.getValue());
+            assertEquals(12, (long) resp.getValue());
         }
     }
 
@@ -213,11 +213,11 @@ public class J4pExecIntegrationTest extends AbstractJ4pIntegrationTest {
                 }
                 request = new J4pExecRequest(cfg,itSetup.getOperationMBean(),"intArguments",10,20);
                 resp = j4pClient.execute(request,type);
-                assertEquals(30, (int) resp.getValue());
+                assertEquals(30, (long) resp.getValue());
 
                 request = new J4pExecRequest(cfg,itSetup.getOperationMBean(),"intArguments",10,null);
                 resp = j4pClient.execute(request,type);
-                assertEquals(-1,(int) resp.getValue());
+                assertEquals(-1, (long) resp.getValue());
 
                 try {
                     request = new J4pExecRequest(cfg,itSetup.getOperationMBean(),"intArguments",null,null);
@@ -282,7 +282,7 @@ public class J4pExecIntegrationTest extends AbstractJ4pIntegrationTest {
                 Map<?, ?> res = resp.getValue();
                 assertEquals(res.get("eins"),"fcn");
                 assertEquals(((List<?>) res.get("zwei")).get(1),"svw");
-                assertEquals(res.get("drei"),10);
+                assertEquals(res.get("drei"),10L);
                 assertEquals(res.get("vier"),true);
             }
 

--- a/client/java/src/test/java/org/jolokia/client/request/J4pReadIntegrationTest.java
+++ b/client/java/src/test/java/org/jolokia/client/request/J4pReadIntegrationTest.java
@@ -215,7 +215,7 @@ public class J4pReadIntegrationTest extends AbstractJ4pIntegrationTest {
             assertTrue(allAttrs.contains("Name"));
             assertTrue(allAttrs.contains("Bytes"));
 
-            Integer val = resp.getValue(new ObjectName(itSetup.getAttributeMBean()),"MemoryUsed");
+            Long val = resp.getValue(new ObjectName(itSetup.getAttributeMBean()),"MemoryUsed");
             assertNotNull(val);
 
             try {
@@ -225,7 +225,7 @@ public class J4pReadIntegrationTest extends AbstractJ4pIntegrationTest {
                 assertTrue(exp.getMessage().contains("Aufsteiger"));
             }
 
-            Integer bytes = resp.getValue("Bytes");
+            Long bytes = resp.getValue("Bytes");
             assertNotNull(bytes);
 
             try {
@@ -304,13 +304,13 @@ public class J4pReadIntegrationTest extends AbstractJ4pIntegrationTest {
         for (J4pReadRequest request  : readRequests("jolokia.it:type=mxbean","ComplexTestData")) {
             J4pReadResponse response = j4pClient.execute(request);
             JSONObject value = response.getValue();
-            assertEquals(value.get("number"),1968);
+            assertEquals(value.get("number"),1968L);
             assertEquals(value.get("string"),"late");
 
             List<?> set = (List<?>) value.get("set");
             assertEquals(set.size(),2);
-            assertTrue(set.contains(12));
-            assertTrue(set.contains(14));
+            assertTrue(set.contains(12L));
+            assertTrue(set.contains(14L));
 
             Map<?, ?> map = (Map<?, ?>) value.get("map");
             assertEquals(map.size(),2);

--- a/client/java/src/test/java/org/jolokia/client/request/J4pWriteIntegrationTest.java
+++ b/client/java/src/test/java/org/jolokia/client/request/J4pWriteIntegrationTest.java
@@ -46,17 +46,17 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
 
     @Test
     public void simple() throws MalformedObjectNameException, J4pException {
-        checkWrite("IntValue",null,42);
+        checkWrite("IntValue",null,42L);
     }
 
     @Test
     public void withPath() throws MalformedObjectNameException, J4pException {
-        checkWrite("ComplexNestedValue","Blub/1/numbers/0",13);
+        checkWrite("ComplexNestedValue","Blub/1/numbers/0",13L);
     }
 
     @Test
     public void withBeanPath() throws MalformedObjectNameException, J4pException {
-        checkWrite("Bean","value",41);
+        checkWrite("Bean","value",41L);
     }
 
     @Test
@@ -74,7 +74,7 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
         Map<String, Object> map = createTestMap();
         checkWrite(new String[]{"POST"}, "Map", null, map);
         checkWrite("Map","fcn","svw");
-        checkWrite("Map","zahl",20);
+        checkWrite("Map","zahl",20L);
 
         // Write an not yet known key
         J4pWriteRequest wReq = new J4pWriteRequest(IT_ATTRIBUTE_MBEAN,"Map","hofstadter","douglas");
@@ -92,7 +92,7 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
         map.put("zwei", "bvb");
         map.put("drei", true);
         map.put("vier", null);
-        map.put("fuenf", 12);
+        map.put("fuenf", 12L);
         return map;
     }
 
@@ -100,14 +100,14 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
     void list() throws MalformedObjectNameException, J4pException {
         List<Object> list = new ArrayList<>();
         list.add("fcn");
-        list.add(42);
+        list.add(42L);
         list.add(createTestMap());
         list.add(null);
         list.add(new BigDecimal("23.2"));
         checkWrite(new String[] { "POST" }, "List",null,list);
         checkWrite("List","0",null);
         checkWrite("List","0","");
-        checkWrite("List","2",42);
+        checkWrite("List","2",42L);
     }
 
 
@@ -129,7 +129,7 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
 
     @Test
     public void array2D() throws MalformedObjectNameException, J4pException {
-        final int[][] input = new int[][] { { 1, 2 }, { 3, 4 } };
+        final long[][] input = new long[][] { { 1, 2 }, { 3, 4 } };
         checkWrite(new String[] { "POST" }, "Array2D", null, input, resp -> {
             JSONArray val = resp.getValue();
             assertEquals(val.size(), input.length);
@@ -169,7 +169,7 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
 
     @Test
     public void mxNumbers() throws MalformedObjectNameException, J4pException {
-        final Integer[] input = { 1,2 };
+        final Long[] input = { 1L, 2L };
         checkMxWrite("Numbers",null,input, resp -> {
             JSONArray val = resp.getValue();
             assertEquals(val.size(), input.length);
@@ -181,9 +181,9 @@ public class J4pWriteIntegrationTest extends AbstractJ4pIntegrationTest {
 
     @Test
     public void mxMap() throws MalformedObjectNameException, J4pException {
-        final Map<String,Integer> input = new HashMap<>();
-        input.put("roland",13);
-        input.put("heino",19);
+        final Map<String,Long> input = new HashMap<>();
+        input.put("roland",13L);
+        input.put("heino",19L);
         checkMxWrite(new String[] {"POST"},"Map", null, input, resp -> {
             JSONObject val = resp.getValue();
             assertEquals(val.size(), input.size());

--- a/json/src/main/java/org/jolokia/json/parser/JSONParser.java
+++ b/json/src/main/java/org/jolokia/json/parser/JSONParser.java
@@ -410,9 +410,9 @@ public class JSONParser {
 //        if (v.bitLength() <= 15) {
 //            return v.shortValue();
 //        }
-        if (v.bitLength() <= 31) {
-            return v.intValue();
-        }
+//        if (v.bitLength() <= 31) {
+//            return v.intValue();
+//        }
         if (v.bitLength() <= 63) {
             return v.longValue();
         }

--- a/json/src/test/java/org/jolokia/json/parser/ParserArraysTest.java
+++ b/json/src/test/java/org/jolokia/json/parser/ParserArraysTest.java
@@ -105,7 +105,7 @@ public class ParserArraysTest {
         assertEquals(array.get(1), new BigDecimal("1.4"));
         assertEquals(array.get(2).toString(), "9223372036854775808");
         assertEquals(array.get(3), 9223372036854775806L);
-        assertEquals(array.get(4), 42);
+        assertEquals(array.get(4), 42L);
         assertTrue((Boolean) array.get(5));
         assertNull(array.get(6));
     }

--- a/json/src/test/java/org/jolokia/json/parser/ParserObjectsTest.java
+++ b/json/src/test/java/org/jolokia/json/parser/ParserObjectsTest.java
@@ -120,7 +120,7 @@ public class ParserObjectsTest {
         assertEquals(object.get("keyBigDecimal"), new BigDecimal("1.4"));
         assertEquals(object.get("keyBigInteger").toString(), "9223372036854775808");
         assertEquals(object.get("keyLong"), 9223372036854775806L);
-        assertEquals(object.get("keyInteger"), 42);
+        assertEquals(object.get("keyInteger"), 42L);
         assertTrue((Boolean) object.get("keyBoolean"));
         assertTrue(object.containsKey("keyNull"));
         assertNull(object.get("keyNull"));

--- a/json/src/test/java/org/jolokia/json/parser/ParserValuesTest.java
+++ b/json/src/test/java/org/jolokia/json/parser/ParserValuesTest.java
@@ -96,10 +96,8 @@ public class ParserValuesTest {
     @Test
     public void parseInteger() throws ParseException, IOException {
         Number value = (Number) new JSONParser().parse(new BigInteger(Integer.toString(Integer.MAX_VALUE)).subtract(BigInteger.TWO).toString());
-        assertTrue(value instanceof Integer);
-        assertEquals((int) value, 2147483645);
-        //noinspection RedundantCast
-        assertEquals((long) (int) value, 2147483645L);
+        assertTrue(value instanceof Long);
+        assertEquals((long) value, 2147483645L);
     }
 
 }

--- a/support/jmx/src/test/java/org/jolokia/support/jmx/JolokiaMBeanServerHandlerTest.java
+++ b/support/jmx/src/test/java/org/jolokia/support/jmx/JolokiaMBeanServerHandlerTest.java
@@ -73,7 +73,7 @@ public class JolokiaMBeanServerHandlerTest {
             String chiliS = (String) pServer.getAttribute(oName,"Chili");
             JSONObject chiliJ = new JSONParser().parse(chiliS, JSONObject.class);
             assertEquals(chiliJ.get("name"), "Bhut Jolokia");
-            assertEquals(chiliJ.get("scoville"), 1000000);
+            assertEquals(chiliJ.get("scoville"), 1000000L);
 
             server.unregisterMBean(oName);
             Assert.assertFalse(pServer.isRegistered(oName));

--- a/support/jmx/src/test/java/org/jolokia/support/jmx/JsonDymamicMBeanImplTest.java
+++ b/support/jmx/src/test/java/org/jolokia/support/jmx/JsonDymamicMBeanImplTest.java
@@ -140,8 +140,8 @@ public class JsonDymamicMBeanImplTest {
         platformServer.setAttribute(testName,new Attribute("Numbers","8,15"));
         String nums = (String) platformServer.getAttribute(testName,"Numbers");
         JSONArray numsJ = toJSON(nums, JSONArray.class);
-        assertEquals(numsJ.get(0), 8);
-        assertEquals(numsJ.get(1), 15);
+        assertEquals(numsJ.get(0), 8L);
+        assertEquals(numsJ.get(1), 15L);
         assertEquals(numsJ.size(), 2);
     }
 
@@ -173,9 +173,9 @@ public class JsonDymamicMBeanImplTest {
         assertEquals(chili.getValue(), "aji");
 
         JSONArray numsJ = toJSON((String) num.getValue(), JSONArray.class);
-        assertEquals(numsJ.get(0), 16);
-        assertEquals(numsJ.get(1), 11);
-        assertEquals(numsJ.get(2), 68);
+        assertEquals(numsJ.get(0), 16L);
+        assertEquals(numsJ.get(1), 11L);
+        assertEquals(numsJ.get(2), 68L);
         assertEquals(numsJ.size(), 3);
 
         assertEquals(platformServer.getAttributes(testName, new String[0]).size(), 0);

